src/nccl_ofi_scheduler.c

```diff
@@ -92,8 +136,12 @@ static inline int set_schedule_by_threshold(nccl_net_ofi_threshold_scheduler_t *
 {
 	int ret = 0;
 	if (size > scheduler->rr_threshold) {
-		nccl_net_ofi_set_multiplexing_schedule(size, num_rails,
-						       align, schedule);
+		if (size > scheduler->mux_rr_threshold){
+			nccl_net_ofi_set_multiplexing_schedule(size, num_rails, align, schedule);
+		}
+		else {
+			nccl_net_ofi_set_multiplexed_round_robin_schedule(scheduler, size, num_rails, align, schedule);
+		}
 	} else {
 		ret = set_round_robin_schedule(scheduler, size, num_rails, schedule);
 	}
```

Comment:
Just a personal opinion: this logic would be clearer as
```
if (size < rr_threshold) {
   do_rr
} else if (size < mux_rr_threshold) {
   do_mux_rr
} else {
   do_mux
}
```

Reply:
Since we now schedule based on the min_stripe_size we can ignore this :)